README.md

```diff
@@ -36,71 +36,68 @@ using System.Diagnostics;
 using System.Linq;
 using System.Threading.Tasks;
 
-class Demo
+public class Demo
 {
 	private readonly PairedNuimoManager _pairedNuimoManager = new PairedNuimoManager();
 	private INuimoController _nuimoController;
 
-	private async Task GetPairedNuimos()
+	private async void GetPairedNuimos()
 	{
 		var nuimoControllers = await _pairedNuimoManager.ListPairedNuimosAsync();
 		_nuimoController = nuimoControllers.ElementAt(0);
 	}
 
-	private async Task Connect()
+	private async void Connect()
 	{
 		var isConnected = await _nuimoController.ConnectAsync();
 	}
 
 	private void AddDelegates()
 	{
-		_nuimoController.GestureEventOccurred     += OnNuimoGestureEvent;
-		_nuimoController.FirmwareVersionRead      += OnFirmwareVersion;
-		_nuimoController.ConnectionStateChanged   += OnConnectionState;
+		_nuimoController.GestureEventOccurred += OnNuimoGestureEvent;
+		_nuimoController.FirmwareVersionRead += OnFirmwareVersion;
+		_nuimoController.ConnectionStateChanged += OnConnectionState;
 		_nuimoController.BatteryPercentageChanged += OnBatteryPercentage;
-		_nuimoController.LedMatrixDisplayed       += OnLedMatrixDisplayed;
+		_nuimoController.LedMatrixDisplayed += OnLedMatrixDisplayed;
 	}
 
-	private void OnNuimoGestureEvent(NuimoGestureEvent nuimoGestureEvent)
+	private void OnNuimoGestureEvent(INuimoController controller, NuimoGestureEvent nuimoGestureEvent)
 	{
 		Debug.WriteLine("Event: " + nuimoGestureEvent.Gesture + ", " + nuimoGestureEvent.Value);
 	}
 
-	private void OnFirmwareVersion(string firmwareVersion)
+	private void OnFirmwareVersion(INuimoController controller, string firmwareVersion)
 	{
 		Debug.WriteLine(firmwareVersion);
 	}
 
-	private void OnConnectionState(NuimoConnectionState nuimoConnectionState)
+	private void OnConnectionState(INuimoController controller, NuimoConnectionState nuimoConnectionState)
 	{
 		Debug.WriteLine("Connection state: " + nuimoConnectionState);
+
+		var displayInterval = 5.0;
+		var matrixString = "         " +
+								"         " +
+								" ..   .. " +
+								"   . .   " +
+								"    .    " +
+								"   . .   " +
+								" ..   .. " +
+								"         " +
+								"         ";
```

Comment:
Can you please use the same formatting for the matrix string as in the current README file? This said, first matrix string line starts next line (line 80) and all lines start in the same column.

Additionally, the matrix should only be sent if connection state changes to connected. I suggest to use an early exit:

	private void OnConnectionState(INuimoController controller, NuimoConnectionState nuimoConnectionState)
	{
  		Debug.WriteLine("Connection state: " + nuimoConnectionState);
		if (nuimoConnectionState != NuimoConnectionState.Connected) { return; }
		//...Display matrix here...